src/calculateSimilarity.ts

```diff
@@ -0,0 +1,37 @@
+import { disassembleHangul } from './disassemble';
+
+export function calculateSimilarity(a: string, b: string): number {
```

Comment:
이름이 더 심플하면 좋을 것 같아요. calculate가 없어도 좋을 것 같더라구요. date-fns가 add, format과 같은 이름으로 단순하게 제공하는 것 처럼요

```suggestion
export function similarity(a: string, b: string): number {
```

Reply:
오오 이거 넘 좋은 관점이네요,
한번 고민 후 다시 코맨트 남기겠습니다!

Reply:
처음 제공할때 이름을 확실히 정하고 나가는게 좋을 것 같아서 깊게 고민해보죠!!

Reply:
저는 계산한다는 맥락이 굳이 외부에 노출될 필요가 있는 맥락인가하는 생각이 있어서, `getSimilarity`라는 이름도 충분하다고 생각합니다.

@manudeli 님께서 제안주신 `similarity`는 명사라, `const similarity = 유사도함수()`와 같은 use case에서 함수의 결과가 담길 변수와 이름이 충돌할 수 있다는 생각도 들어요!

Reply:
**getSimilarity** 좋습니다! 간결하고 오해의 소지가 없을 것 같아요

Reply:
@okinawaa @evan-moon 함수들 명명 규칙을 정해보면 좋겠어요. BREAKING CHANGE가 있기는 하겠지만 아직 사용량이 적을 때 명명규칙을 정하고 빠르게 BREAKING CHANGE를 내는 것도 장기적으로 도움이 될 수 있을 거 같아요

disassembleHangul, disassembleHangulToGroup도 disassemble로는 어떨까요? (BREAKING CHANGE 필요) Hangul이라는 맥락을 알필요가 있었다면 getSimilarityHangul이어야 하는지 아닌지 정의할 필요가 있을 거 같아요
```ts
disassemble(korStr, { type: 'string' }) // disassembleHangul
disassemble(korStr, { type: 'array' }) // disassembleHangulToGroup
```

Reply:
좋은 제안인 것 같습니다.
저는 가급적이면 Hangul이라는 맥락이 함수명에 포함되는 것이 좋다고 생각하는데요.

물론 `import { disassemble } from 'es-hangul'`과 같은 구문을 보면 "이 함수가 한글을 분해한다"라는 맥락을 파악할 수는 있겠으나, import 문의 특성 상 모듈의 최상단에 위치하여 함수가 소비되는 곳과의 거리가 멀 가능성이 높다는 점이 그 이유인데요.

이런 상황이라면 개발자가 `disassemble`이라는 추상 레벨이 높은 함수명만 보고 함수의 역할과 동작을 예측하기 어렵기 때문에 좋은 네이밍이 아니라고 생각했어요.

그래서 함수명만으로도 명확하게 함수의 역할과 동작을 예측할 수 있는 `*Hangul`이라는 네이밍이 더 DX에 좋지 않을까 했습니다.

Reply:
> 물론 import { disassemble } from 'es-hangul'과 같은 구문을 보면 "이 함수가 한글을 분해한다"라는 맥락을 파악할 수는 있겠으나, import 문의 특성 상 모듈의 최상단에 위치하여 함수가 소비되는 곳과의 거리가 멀 가능성이 높다는 점이 그 이유인데요.

저도 이 점에 동의해요. 그리고 이에 더해서 date-fns와는 es-hangul이 인자를 받는 데에 다른 특징을 갖고 있다고 생각했어요. date-fns의 add, format는 date라는 맥락이 함수명에 포함되지 않아도 괜찮은 게 인자로 받는 변수명에 그 내용이 포함될 가능성이 높아서라고 생각했어요.

```ts
add(createdAt, { day: 1 }) // 변수명 createdAt에서 Date객체임을 알 수 있음
add('hello', { day: 1 }) // 타입에러! Date객체여야 함
```

es-hangul의 경우 변수명이나 타입적으로 인자가 변수라는 것을 알 수 없기 때문에 Hangul이라는 맥락이 disassembleHangul에 포함되는 것이 더 좋다고 생각했어요

### Bad

```ts
disassemble(user.name) // user.name이 한국어야 한다는 것을 알 수 없음
disassemble('English Name') // 한국어가 아니더라도 타입에러가 발생하지 않음
```

### Good

```ts
disassembleHangul(user.name) // 한국어야 한다는 것을 알 수 없음. 하지만 hangul인지 검사 필요하다는 것을 이해할 수 있음
disassembleHangul('English Name') // 한국어가 아니더라도 타입에러가 발생하지 않음 하지만 이상함을 이해할 수 있음
```

그래서 String이지만 한글이어야하는 인자를 받는 경우에 함수명에 `~~~Hangul`과 같은 suffix가 있으면 하는 바람이 있습니다

Reply:
이슈화 해두었습니다 https://github.com/toss/es-hangul/issues/121